crates/oxc_linter/src/rules/eslint/no_redeclare.rs

```diff
@@ -69,16 +68,18 @@ impl Rule for NoRedeclare {
         Self { built_in_globals }
     }
 
-    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+    fn run_once(&self, ctx: &LintContext) {
         let redeclare_variables = ctx.semantic().redeclare_variables();
 
-        match node.kind() {
-            AstKind::BindingIdentifier(ident) => {
+        for node in ctx.nodes().iter() {
```

Comment:
This is still looping every identifier, then every redeclare_variables.

If you store the symbol_id in your redeclare_variables, then you can lookup the original symbol for its span.